rpcs3/Emu/Cell/Modules/cellMic.cpp

```diff
@@ -258,6 +258,29 @@ inline void microphone_device::variable_byteswap(const void* src, void* dst)
 	}
 }
 
+inline u32 microphone_device::convert_16_bit_pcm_to_float(const std::vector<u8>& buffer, usz num_bytes)
+{
+	constexpr usz float_buf_size = 2 * inbuf_size;
+	static_assert((float_buf_size % sizeof(u32)) == 0);
+
+	float_buf.resize(float_buf_size, 0);
+	ensure(num_bytes * 2 <= float_buf.size());
```

Comment:
Again, less magical:
```suggestion

	const u32 bytes_to_write = static_cast<u32>(num_bytes * (sizeof(f32) / sizeof(s16)));
	ensure(bytes_to_write <= float_buf.size());
```